pkg/controller/storagecluster/external_resources.go

```diff
@@ -161,16 +196,18 @@ func (r *ReconcileStorageCluster) newExternalCephObjectStoreInstances(
 // ensureExternalStorageClusterResources ensures that requested resources for the external cluster
 // being created
 func (r *ReconcileStorageCluster) ensureExternalStorageClusterResources(instance *ocsv1.StorageCluster, reqLogger logr.Logger) error {
-	// check for the status boolean value accepted or not
-	if instance.Status.ExternalSecretFound {
+	if r.sameExternalSecretData(instance) {
 		return nil
 	}
 	err := r.createExternalStorageClusterResources(instance, reqLogger)
 	if err != nil {
 		reqLogger.Error(err, "could not create ExternalStorageClusterResource")
 		return err
 	}
-	instance.Status.ExternalSecretFound = true
+	// as there are no errors, update the external checksum status
+	if externalSecretDataHash, err := r.externalSecretDataChecksum(instance); err == nil {
```

Comment:
I don't think we should be calling `externalSecretDataChecksum()` here, since it is a separate call to Get() the Secret and the data may have changed since the last call. I think you should just remove the `sameExternalSecretData()` function and do the comparison directly in `ensureExternalStorageClusterResources()`. Make sense?

Reply:
Yes, it make sense. I have made a minor change, which should address both the concerns (one in the above comment as well).
@jarrpa , please take a look...